Tensile/KernelWriter.py

```diff
@@ -352,35 +385,41 @@ def makeSchedule(self, kernel, tensorParametersA, tensorParametersB, localWriteE
       if kernel["EnableMatrixInstruction"] and kernel["ScheduleIterAlg"] == 3:
         self.lwStartMfmaIndex = self.lwEndMfmaIndex
     else:
-      # create a plan:
+      #################
+      # create a plan #
+      #################
       itemsLWToSched = list(self.localWriteACode.items()) + list(self.localWriteBCode.items())
-      # piggyback each global read instruction after each local write instruction to maximize G2L buffer utilization
-      if kernel["PrefetchGlobalRead"] == 2 or kernel["DepthULdsDivisor"]>1:
-        itemsLWToSched = list(zip_longest(itemsLWToSched, itemsGRToSchedLater, fillvalue=Code.Module("Placeholder")))
-      try:
-        # count how many items to schedule (local writes as well as global reads)
-        _filterType = [Code.LocalWriteInst, Code.GlobalReadInst]
-        writesToSched = sum(1 for p, q in itemsLWToSched if p.countTypeList(_filterType) or q.countTypeList(_filterType))
-      except TypeError: # itemsLWToSched not tuple -> no piggybacked GR code
-        if 1:
-          # This counts the number of modules which contain a ds_write
-          # Scheduler below keeps all writes in the same module in same iteration
-          # so this is better match to what it is trying to do
-          writesToSched = sum(1 for item in itemsLWToSched if item.countType(Code.LocalWriteInst))
-        else:
-          # count the number of writes, this doesn't match how they are
-          # scheduled so pushes writes up too far
-          writesToSched = self.localWriteACode.countType(Code.LocalWriteInst) + \
-                      self.localWriteBCode.countType(Code.LocalWriteInst)
+      # extend localWrtie by inserting empty Module
+      itemsLWToSchedTemp = []
+      for i in range(len(itemsLWToSched)-1):
+        itemsLWToSchedTemp.append(itemsLWToSched.pop(0))
+        for j in range(99):
+          itemsLWToSchedTemp.append(Code.Module())
+      if itemsLWToSched:
+        itemsLWToSchedTemp.append(itemsLWToSched.pop(0))
+        for i in range(numLocalWritesPerSched + numLocalWritesPerSched % 100 - len(itemsLWToSchedTemp) % numLocalWritesPerSched):
```

Comment:
should we define 100 or 99 as a predefined varaible?

Reply:
done